### PR TITLE
feat: matches block initial commit

### LINKED ISSUE #12: Next matches block
Next matches are retrieved from AEM CS via a persisted graphql. Example call:

[https://publish-p47754-e237356.adobeaemcloud.com/graphql/execute.json/realmadridmastersite/matche[…]BsquadId=3kq9cckrnlogidldtdie2fkbl%3Balang=es-ES](https://publish-p47754-e237356.adobeaemcloud.com/graphql/execute.json/realmadridmastersite/matches%3BfromDate=2023-04-25T12:00%3BtoDate=2023-12-15T12:00%3BsquadId=3kq9cckrnlogidldtdie2fkbl%3Balang=es-ES)

As a particularity and due to a limitation with graphQL and Dynamic Media, the result provides de logo  publish url like:
https://publish-p47754-e237356.adobeaemcloud.com/content/dam/common/statics/public-content/badge/football/3kq9cckrnlogidldtdie2fkbl.png

and we have a servlet that converts that url into the DM url for the asset, you just have to add the ‘app’ selector and any required parameters: :https://publish-p47754-e237356.adobeaemcloud.com/content/dam/common/statics/public-content/badge/football/3kq9cckrnlogidldtdie2fkbl.app.png?wid=150

The block will grab the content from the persisted queries, filter out the matches were Real Madrid is not the `homeTeam` and render them with the same styling than in the current website.
Screenshot 2023-04-25 at 15 55 45
Screenshot 2023-04-25 at 15 51 50

URL where you can see the original look and feel https://www.realmadrid.com/area-vip/partido-vip

### PR DESCRIPTION
**Pull request by vdua**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #12 

Test URLs:
- Before: https://main--realmadrid--hlxsites.hlx.page/drafts/vdua/partido-vip
- After: https://12-matches-block--realmadrid--hlxsites.hlx.page/drafts/vdua/partido-vip

### CODE REVIEW

**Comment by vdua** [comment]
Pending Items

 * working out the hyperlinks
 * showing the add to calendar button and animation associated with it
 * configuring out which months to display.

**Review by chicharr — APPROVED** [review]
Created a ticket for the follow-up tasks https://github.com/hlxsites/realmadrid/issues/65

File: blocks/matches/matches.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+export default async function decorate(block) {
+  const config = readBlockConfig(block);
+  const placeholders = await fetchPlaceholders();
+  const { matchesGqApi, noMatches } = placeholders;
+  const { sport } = config;
+  const url = new URL(`${matchesGqApi}${API[sport.toLowerCase()]}`); // todo: add params fromDate endDate
+  const response = await fetch(url);
+  const data = await response.json();
+  const _items = data.data.matchList.items.concat(data.data.matchList.items);
```
Comment (vdua) [inline blocks/matches/matches.js:137]: doing this to show what happens when the number of matches go beyond the area of the screen. Will remove that before merging

File: blocks/matches/matches.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    </div>
+    <div class="price">${fromText} 1.400€</div>
+    <a class="button cta" href="#">${ctaLabel}</a>
+  </div>
+  `;
+  return div;
+};
+
+function getCurrentSeason() {
+  return '22/23';
```
Comment (vdua) [inline blocks/matches/matches.js:52]: got the information from real-madrid regarding this. will update that.

File: blocks/matches/matches.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    ...items,
+    createDiv('empty-match', noMatches),
+  );
+
+  // const months = new Set(items.map((x) => x.dataset.month));
+  const months = [
+    monthNames[currentMonth - 1],
+    monthNames[currentMonth],
+    monthNames[currentMonth + 1]]; // ideally this would be created
+    // from the api response
```
Comment (vdua) [inline blocks/matches/matches.js:150]: the current api response returns matches for april only, hence doing this just for the demonstration. Will be removed before merging

**Pull request merged**